api/bots_api/bot_lib.py

```diff
@@ -105,6 +111,67 @@ def get_config_info(self, bot_name, section=None):
         config.readfp(open(conf_file_path))  # type: ignore
         return dict(config.items(section))
 
+class EmbeddedBotHandler(object):
+    def __init__(self, client):
+        # type: (Client) -> None
+        # Only expose a subset of our Client's functionality
+        user_profile = client.get_profile()
+        self._rate_limit = RateLimit(20, 5)
+        self._client = client
+        try:
+            self.full_name = user_profile['full_name']
+            self.email = user_profile['email']
+        except KeyError:
+            logging.error('Cannot fetch user profile, make sure you have set'
+                          ' up the zuliprc file correctly.')
+            sys.exit(1)
+
+    def embedded_send_message(self, message):
+        # type: (Dict[str, Any]) -> Dict[str, Any]
+        if self._rate_limit.is_legal():
+            return internal_send_message(realm=message['realm'], sender_email=message['sender'],
+                                         recipient_type_name=message['type'], recipients=message['to'],
+                                         subject=message['subject'], content=message['content'])
+        else:
+            self._rate_limit.show_error_and_exit()
+
+    def update_message(self, message):
+        # type: (Dict[str, Any]) -> Dict[str, Any]
+        if self._rate_limit.is_legal():
+            return self._client.update_message(message)
+        else:
+            self._rate_limit.show_error_and_exit()
+
+    def send_reply(self, message, response):
+        # type: (Dict[str, Any], str) -> Dict[str, Any]
+        if message['type'] == 'private':
+            return self.send_message(dict(
+                sender_email=message['sender'],
+                realm=message['sender'].realm,
+                type='private',
+                to=[x['email'] for x in message['display_recipient'] if self.email != x['email']],
+                subject=message['subject'],
+                content=response,
+            ))
+        else:
+            return self.send_message(dict(
+                sender_email=message['sender'],
+                type='stream',
+                realm=message['sender'].realm,
+                to=message['display_recipient'],
+                subject=message['subject'],
+                content=response,
+            ))
+
+    def get_config_info(self, bot_name, section=None):
```

Comment:
@robot-dreams not exactly, but I do think we'll be needing a source to store configuration details for bots. 

What are your thoughts?